File: app/src/main/java/com/borisruzanov/russianwives/mvp/model/interactor/ChatsInteractor.java
package com.borisruzanov.russianwives.mvp.model.interactor;

import com.borisruzanov.russianwives.mvp.model.repository.FirebaseRepository;
import com.borisruzanov.russianwives.utils.UserChatListCallback;

public class ChatsInteractor {

  private FirebaseRepository repository;

  public ChatsInteractor(FirebaseRepository repository) {
    this.repository = repository;
  }

  public void getUsersChats(UserChatListCallback callback){
    repository.createUserChats(callback);
  }

}

File: app/src/main/java/com/borisruzanov/russianwives/mvp/model/interactor/chats/ChatsInteractor.kt
package com.borisruzanov.russianwives.mvp.model.interactor.chats

import com.borisruzanov.russianwives.mvp.model.repository.chats.ChatsRepository
import com.borisruzanov.russianwives.utils.UserChatListCallback
import javax.inject.Inject

class ChatsInteractor @Inject constructor(private val repository: ChatsRepository) {

    fun getUsersChats(callback: UserChatListCallback) {
        repository.getChats(callback)
    }

}
